src/test/java/bio/terra/stairway/ScenarioTest.java

```diff
@@ -715,4 +718,109 @@ private String makeExistingFile() throws Exception {
   private String makeFilename() {
     return "/tmp/test." + UUID.randomUUID().toString() + ".txt";
   }
+
+  /**
+   * Class to enable creating Flights and waiting for their completion in background threads,
+   * caching results and any thrown exceptions for inspection in the main test thread.
+   */
+  private class ThreadedTestRunnable implements Runnable {
+
+    Phaser phaser;
+    private final int index;
+    private final String filename;
+    Exception exception;
+    FlightState result;
+
+    ThreadedTestRunnable(Phaser phaser, int index) {
+      this.phaser = phaser;
+      this.index = index;
+      this.filename = makeFilename();
+    }
+
+    @Override
+    public void run() {
+
+      // This basically implements the same logic as simpleTest in a background thread, holding
+      // results and any caught exceptions for inspection by the main thread.  A synchronization
+      // point is also injected via the use of a Phaser to introduce a barrier before all threads
+      // submit their flights, with the intent of triggering timestamp collisions to regression test
+      // the fix for PF-785.
+
+      try {
+        // Submit the test flight
+        FlightMap inputParameters = new FlightMap();
+        inputParameters.put("filename", filename);
+        inputParameters.put("text", "testing 1 2 3");
+
+        logger.debug(
+            "Waiting to launch TestFlight '{}' with filename '{}'", getFlightId(), filename);
+
+        // Phaser is used here as a simple barrier to try to induce duplicate timestamps.  All
+        // threads wait on the barrier until the last one reaches it.
+        phaser.arriveAndAwaitAdvance();
```

Comment:
How did I not know about Phaser?! That is really cool.
I should just know that any possible problem I have in Java has already been solved. :)